File: Problems/Algorithms/62. Unique Paths/unique_paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        def fact(n: int) -> int:
            if n == 0 or n == 1:
                return 1;
            return n * fact(n-1)

        return fact(m+n-2) // (fact(m-1) * fact(n-1))

File: Problems/Algorithms/62. Unique Paths/unique-paths.cpp
#include <algorithm>

using namespace std;

class Solution {
public:
    int uniquePaths(int m, int n) {
        if (n == 1) {
            return 1;
        }

        unsigned long long ans = 1LL;
        for (unsigned i = 1; i <= min(n-1, m-1); i++) {
            ans = ans * (n + m - 2 - min(n-1, m-1) + i) / i;
        }

        return ans;
    }
};
